runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityOverlay.java

```diff
@@ -112,19 +113,22 @@ public Dimension render(Graphics2D graphics)
 
 		});
 
-		if (markOfGrace != null && config.highlightMarks())
+		if (marksOfGrace != null && marksOfGrace.size() > 0 && config.highlightMarks())
```

Comment:
This will not be null ever now that you made it a list and initialized it. This can just be `if (config.highlightMarks() && !marksOfGrace.isEmpty())`
Might not even need the empty check as looping over a list with 0 entries will not do anything, I am not sure what is best pracitce.